project/Version.scala

```diff
@@ -30,4 +30,26 @@ object Version {
       Seq(file)
     }.taskValue
   )
+
+  lazy val generateVersionClass = Seq(
+    sourceGenerators in Compile += Def.task {
+      val file = (sourceManaged in Compile).value / "com" / "commercetools" / "sunrise" / "common" / "version" / "SunriseSolutionInfo.java"
+      IO.write(file,
+        s"""package com.commercetools.sunrise.common.version;
+          |
+          |import io.sphere.sdk.client.SolutionInfo;
+          |
+          |public class SunriseSolutionInfo extends SolutionInfo {
+          |
+          |
+          |    public SunriseSolutionInfo() {
+          |        setName("sunrise-java-shop-framework");
+          |        setVersion("${version.value}");
+          |    }
+          |
+          |}
+          |""".stripMargin)
+      Seq(file)
+    }.taskValue)
+
```

Comment:
It's good that you found a viable solution 👍 
I found a possible option that would make the code nicer, using an SBT plugin that allows to access SBT vars from the source code: https://github.com/sbt/sbt-buildinfo
That would great if it works well, and it would even allow us to test everything. Looks promising, don't you think? :)